File: program_data/github_java_sort_function/6/26.java
  private static int[] sort(char[] a) {
        int n = a.length;
        int[] order = new int[n];
        int[] count = new int[R + 1]; 
        char[] sorted = new char[n];
        
        for (int i = 0; i < n; i++) {
            count[((int) a[i]) + 1]++;
        }
        
        for (int i = 1; i <= R; i++) {
            count[i] += count[i - 1];
        }
        
        
        for (int i = 0; i < n; i++) {
            order[count[(int) a[i]]] = i;
            sorted[count[(int) a[i]]++] = a[i];            
        }
        
        for (int i = 0; i < n; i++) {
            a[i] = sorted[i];
        }        
        return order;
    }
    
    private static void sort(char[] a, int[] b) {
        assert(a.length == b.length);
        int n = a.length;
        int[] order = sort(a);
        int[] tmp = new int[n];
        for (int i = 0; i < n; i++) {
            tmp[i] = b[order[i]];
        }
        for (int i = 0; i < n; i++) {
            b[i] = tmp[i];
        }        
    }

File: program_data/PKU_raw/12/26.c

int main()
{
	int i, j, n, count;
	float a[20];
	while(1)
	{
		scanf("%f", &a[0]);
		if (a[0] == -1)
			break;
		n=1;
		while(1)
		{
			scanf("%f", &a[n]);
			if (a[n] == 0)
				break;
			n++;
		}
		count=0;
		for (i=0; i<n; i++)
		{
			for (j=i+1; j<n; j++)
			{
				if ((a[i]/a[j]==2.0) || (a[j]/a[i]==2.0))
					count++;
			}
		}
		printf("%d\n", count);
	}

	return 0;
}